pennylane/operation.py

```diff
@@ -1112,27 +1111,37 @@ class Tensor(Observable):
     par_domain = None
 
     def __init__(self, *args):  # pylint: disable=super-init-not-called
-
         self._eigvals_cache = None
         self.obs = []
+        self._args = args
+        self.queue(init=True)
 
-        for o in args:
-            if isinstance(o, Tensor):
-                self.obs.extend(o.obs)
-            elif isinstance(o, Observable):
-                self.obs.append(o)
-            else:
-                raise ValueError("Can only perform tensor products between observables.")
+    def queue(self, context=qml.QueuingContext, init=False):  # pylint: disable=arguments-differ
```

Comment:
I'm not sure I understand the `init` argument?

Reply:
Ah - the queuing logic was previous occuring at the exact same time as the `Tensor` datastructure was being computed.

That is, within `__init__`, there was a single for loop that did two things:

- It queued each constituent observable
- It created the internal `obs` list of constituent observables.

I originally split it into two separate for loops in order to separate out the logic, but realized that it would result in looping through the observable list twice. Which is fine... but as the number of qubits increases, this could be a bottleneck.

So I combined it back into a single for loop, but use `init=True` to ensure that building the datastructure only happens on instantiation, while queuing happens every time.